pallets/xyk/src/lib.rs

```diff
@@ -117,12 +130,12 @@ decl_module! {
             );
 
             ensure!(
-                <assets::Module<T>>::balance(first_asset_id, sender.clone()) >= first_asset_amount,
+                T::Currency::free_balance(first_asset_id, &sender) >= first_asset_amount,
                 Error::<T>::NotEnoughAssets,
             );
 
             ensure!(
-                <assets::Module<T>>::balance(second_asset_id, sender.clone()) >= second_asset_amount,
+                T::Currency::free_balance(second_asset_id, &sender) >= second_asset_amount,
```

Comment:
why not use ensure_can_withdraw? as it also checks for frozen amount.

Reply:
and these checks have to made even though transfer itslef calls ensure_can_withdraw, as we are making two calls to transfer and if one passes the other shouldn't fail. So we need to ensure that both pass.